### ISSUE TITLE
Problem: 0-10V Spindle and dust control trigger for a few seconds at startup

### ISSUE THREAD

**Issue by ColPaulR** [bdring/FluidNC#1073]
### Wiki Search Terms

I2SO, spindle, mist

### Controller Board

MKS Tinybee with external interface circuitry 

### Machine Description

Gantry router/laser. Queenbee Pro 1500x1500 with 2.2kw water cooled spindle, diode laser, and dust collection system. 

### Input Circuits

```yaml
Problem is with output circuitry.
```

### Configuration file

```yaml
board: MKS TinyBee V1.0_001
name: CNC Router XYYZA 0-10V spindle and laser
meta: (01.02.2022) by Skorpi

kinematics:
  Cartesian:

stepping:
  engine: I2S_STATIC
  idle_ms: 255
  pulse_us: 4
  dir_delay_us: 1
  disable_delay_us: 0

axes:
  x:
    # X
    steps_per_mm: 250
    max_rate_mm_per_min: 7500.000
    acceleration_mm_per_sec2: 200.000
    max_travel_mm: 1300.000
    soft_limits: true
    homing:
      cycle: 3
      positive_direction: true
      mpos_mm: 1265.00
      feed_mm_per_min: 300.000
      seek_mm_per_min: 2000.000
      settle_ms: 250
      seek_scaler: 1.100
      feed_scaler: 1.100

    motor0:
      limit_neg_pin: GPIO.33:low
      hard_limits: true
      pulloff_mm: 2.500
      standard_stepper:
        step_pin: I2SO.1
        direction_pin: I2SO.2:low
        disable_pin: I2SO.0

  y:
    # Y
    steps_per_mm: 250
    max_rate_mm_per_min: 7500.000
    acceleration_mm_per_sec2: 200.000
    max_travel_mm: 1300.000
    soft_limits: true
    homing:
      cycle: 2
      positive_direction: false
      mpos_mm: 0.000
      feed_mm_per_min: 300.000
      seek_mm_per_min: 2000.000
      settle_ms: 250
      seek_scaler: 1.100
      feed_scaler: 1.100

    motor0:
      limit_neg_pin: GPIO.32:low
      hard_limits: true
      pulloff_mm: 2.000
      standard_stepper:
        step_pin: I2SO.4
        direction_pin: I2SO.5:low
        disable_pin: I2SO.3
    motor1:
      limit_neg_pin: GPIO.22:low
      hard_limits: true
      pulloff_mm: 1.90
      standard_stepper:
        step_pin: I2SO.7
        direction_pin: I2SO.8:low
        disable_pin: I2SO.6
 
  z:
    # Z
    steps_per_mm: 250
    max_rate_mm_per_min: 7500.000
    acceleration_mm_per_sec2: 200.000
    max_travel_mm: 300.000
    soft_limits: false
    homing:
      cycle: 1
      positive_direction: true
      mpos_mm: 0.000
      feed_mm_per_min: 100.000
      seek_mm_per_min: 1000.000
      settle_ms: 500
      seek_scaler: 1.100
      feed_scaler: 1.100

    motor0:
      limit_neg_pin: GPIO.35:low
      hard_limits: true
      pulloff_mm: 1.500
      standard_stepper:
        step_pin: I2SO.10
        direction_pin: I2SO.11
        disable_pin: I2SO.9

  a:
    # Rotary
    steps_per_mm: 33.333333333333333333333333333333333333333333333333
    max_rate_mm_per_min: 18000.000
    acceleration_mm_per_sec2: 1500.000
    max_travel_mm: 325.000
    soft_limits: false

    motor0:
      standard_stepper:
        step_pin: I2SO.13
        direction_pin: I2SO.14
        disable_pin: I2SO.12

i2so:
  bck_pin: GPIO.25
  data_pin: GPIO.27
  ws_pin: GPIO.26

spi:
  miso_pin: GPIO.19
  mosi_pin: GPIO.23
  sck_pin: GPIO.18

sdcard:
  cs_pin: GPIO.5
  card_detect_pin: GPIO.34
  
control:
  safety_door_pin: NO_PIN
  reset_pin: NO_PIN
  # feed_hold_pin: GPIO.39:low
  cycle_start_pin: NO_PIN
  macro0_pin: NO_PIN
  macro1_pin: NO_PIN
  macro2_pin: NO_PIN
  macro3_pin: NO_PIN

10V:
  # BEEPER Pin 1 on EXP1
  forward_pin: I2SO.21
  pwm_hz: 5000
  # GPIO.21 is LCD_EN Pin 3 on EXP1
  output_pin: GPIO.21   
  enable_pin: NO_PIN
  direction_pin: NO_PIN
  disable_with_s0: true
  s0_with_disable: true
  spinup_ms: 6000
  spindown_ms: 6000
  tool_num: 0
  speed_map: 0=0.000% 4000=0.000% 24000=50.000%
  off_on_alarm: true

Laser:
  pwm_hz: 5000
  output_pin: GPIO.4
  enable_pin: NO_PIN
  disable_with_s0: false
  s0_with_disable: true
  tool_num: 100
  speed_map: 0=0.000% 1000=100.000%
  off_on_alarm: true

macros:
  startup_line0:
  startup_line1:
  macro0: 
  macro1:
  macro2:
  macro3:

coolant:
  # flood_pin: I2SO.20
  mist_pin: I2SO.19
  delay_ms: 1000

probe:
  pin: GPIO.13:low:pu
  check_mode_start: false

start:
  must_home: false

verbose_errors: true
use_line_numbers: true

 # USB  TX0(1)  RX0(3)
 # J3-> TX2(17) RX2(16)
 # J1-> I2SO.23 I2SO.22
 
 # HE0(I2SO.17)   HE1(I2SO.18)   HBED(I2SO.16)
 # FAN2(I2SO.20)
 #                   EXP1                                   EXP2
 #(BEEPER) I2SO.21 | 1  2 | 13 (BTN_ENC)    (SPI MISO) 19 | 1  2 | 18 (SPI SCK)
 #    (LCD_EN)  21 | 3  4 |  4 (LCD_RS)      (BTN_EN1) 14 | 3  4 |  5 (SPI CS)
 #    (LCD_D4)   0 | 5  6   16 (LCD_D5)      (BTN_EN2) 12 | 5  6   23 (SPI MOSI)
 #    (LCD_D6)  15 | 7  8 | 17 (LCD_D7)      (SPI_DET) 34 | 7  8 | RESET
 #             GND | 9 10 | 5V                        GND | 9 10 | 3.3V
 #
 #                            EXP1
 # SPINDLE FORWARD  I2SO.21 | 1  2 | 13 PROBE
 # SPINDLE OUTPUT       21  | 3  4 |  4 LASER OUTPUT
 # UNUSED OUTPUT         0  | 5  6   16 RXD2    
 # UNUSED OUTPUT        15  | 7  8 | 17 TXD2 
 #                      GND | 9 10 | 5V               
 #
 #                              EXP2
 # UNAVAILABLE  (SPI MISO)  19 | 1  2 | 18 (SPI SCK)  UNAVAILABLE
 # UNUSED IO    (BTN_EN1)   14 | 3  4 |  5 (SPI CS)   UNAVAILABLE
 # UNUSED IO    (BTN_EN2)   12 | 5  6 | 23 (SPI MOSI) UNAVAILABLE
 # UNAVAILABLE  (SPI_DET)   34 | 7  8 | RESET         UNAVAILABLE
 #                         GND | 9 10 | 3.3V                     
 #
 #        J40
 # 1 - +5V
 # 2 - GND
 # 3 - MT_DET - GPIO.35 with capacitance
 #
 #        J41
 # 1 - 3D_TOUCH - GPIO.2 OUTPUT   
 # 2 - +5V
 # 3 - GND
 #
 #        TB1
 # 1 - GND
 # 2 - TB - SENSOR VN - GPIO.39  with capacitance
 #
 #        TH1
 # 1 - GND
 # 2 - TH1 - SENSOR VP - GPIO.36 with capacitance
 #
 #        TH2
 # 1 - GND
 # 2 - TH1 - SENSOR VP - UNAVAILABLE
 #
 #        J11
 # 1 - FAN1 - I2SO.19
 # 2 - 24VDC
```

### Startup Messages

```text
ets Jun  8 2016 00:22:57

rst:0x1 (POWERON_RESET),boot:0x1b (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1184
load:0x40078000,len:13220
ho 0 tail 12 room 4
load:0x40080400,len:3028
entry 0x400805e4

[MSG:INFO: FluidNC v3.7.9 https://github.com/bdring/FluidNC]
[MSG:INFO: Compiled with ESP32 SDK:v4.4.4]
[MSG:INFO: Local filesystem type is spiffs]
[MSG:INFO: Configuration file:XYZAB_laser_27Nov23.yaml]
[MSG:DBG: Running after-parse tasks]
[MSG:DBG: Checking configuration]
[MSG:INFO: Machine CNC Router XYYZA 0-10V spindle and laser]
[MSG:INFO: Board MKS TinyBee V1.0_001]
[MSG:INFO: I2SO BCK:gpio.25 WS:gpio.26 DATA:gpio.27]
[MSG:INFO: SPI SCK:gpio.18 MOSI:gpio.23 MISO:gpio.19]
[MSG:INFO: SD Card cs_pin:gpio.5 detect:gpio.34 freq:8000000]
[MSG:INFO: Stepping:I2S_static Pulse:4us Dsbl Delay:0us Dir Delay:1us Idle Delay:255ms]
[MSG:INFO: Axis count 4]
[MSG:INFO: Axis X (-35.000,1265.000)]
[MSG:INFO:   Motor0]
[MSG:INFO:     standard_stepper Step:I2SO.1 Dir:I2SO.2:low Disable:I2SO.0]
[MSG:INFO:  X Neg Limit gpio.33:low]
[MSG:DBG:  X Neg Limit 0]
[MSG:INFO: Axis Y (0.000,1300.000)]
[MSG:INFO:   Motor0]
[MSG:INFO:     standard_stepper Step:I2SO.4 Dir:I2SO.5:low Disable:I2SO.3]
[MSG:INFO:  Y Neg Limit gpio.32:low]
[MSG:DBG:  Y Neg Limit 0]
[MSG:INFO:   Motor1]
[MSG:INFO:     standard_stepper Step:I2SO.7 Dir:I2SO.8:low Disable:I2SO.6]
[MSG:INFO:  Y2 Neg Limit gpio.22:low]
[MSG:DBG:  Y2 Neg Limit 0]
[MSG:INFO: Axis Z (-300.000,0.000)]
[MSG:INFO:   Motor0]
[MSG:INFO:     standard_stepper Step:I2SO.10 Dir:I2SO.11 Disable:I2SO.9]
[MSG:INFO:  Z Neg Limit gpio.35:low]
[MSG:DBG:  Z Neg Limit 0]
[MSG:INFO: Axis A (-325.000,0.000)]
[MSG:INFO:   Motor0]
[MSG:INFO:     standard_stepper Step:I2SO.13 Dir:I2SO.14 Disable:I2SO.12]
[MSG:INFO: Kinematic system: Cartesian]
[MSG:INFO: 10V Spindle Ena:NO_PIN Out:gpio.21 Dir:NO_PIN Fwd:I2SO.21 Rev:NO_PIN Freq:5000Hz Period:8191]
[MSG:INFO: Laser Ena:NO_PIN Out:gpio.4 Freq:5000Hz Period:8191]
[MSG:INFO: Using spindle 10V]
[MSG:INFO: Mist coolant I2SO.19]
[MSG:INFO: Probe Pin: gpio.13:low:pu]
[MSG:INFO: Connecting to STA SSID:RounsavallWPA2]
[MSG:INFO: Connecting.]
[MSG:INFO: Connecting..]
[MSG:INFO: Connecting...]
[MSG:INFO: Connected - IP is 192.168.0.235]
[MSG:INFO: WiFi on]
[MSG:INFO: Start mDNS with hostname:http://fluidnc.local/]
[MSG:INFO: SSDP Started]
[MSG:INFO: HTTP started on port 80]
[MSG:DBG: XYZAB_laser_27Nov23.yaml hash "4CB3DBF2D4F8CCFF9F624B9A0EED6729D5C2FCACC0D1EE53C5C5CDBC57E0A043"]
[MSG:DBG: preferences.json hash "FAAA43D3F2445A758949809DF10CE92AB000B287697AE606894355EAC100FEA4"]
[MSG:DBG: index.html.gz hash "3B628023FBEBA2C44FCC0CD494849DE07F5D327C98AD70F5DF5B2CD374B0DD81"]
[MSG:INFO: Telnet started on port 23]
[MSG:DBG:  Z Neg Limit 0]

Grbl 3.7 [FluidNC v3.7.9 (wifi) '$' for help]
[MSG:DBG:  X Neg Limit 0]
[MSG:DBG:  Y Neg Limit 0]
[MSG:DBG:  Y2 Neg Limit 0]
[MSG:DBG: WebSocket 0 from 192.168.0.230 uri /]
```

### User Interface Software

Happens with all software (CNC3D commander, fluidterm, Lightburn, CNCjs, web interface)

### What happened?

When resetting the controller (MKS Tinybee), both the spindle and the dust collection turn on for a few seconds while the controller is initializing/sending the startup messages. Then both the spindle and dust collection turn off. This does not happen during power on but only when resetting ($BYE or ^R).

The FWD and DCM lines on the Huanyang VFD is connected to the a dual relay module (identical to [https://www.amazon.com/Channel-Optocoupler-Trigger-Expansion-Arduino/dp/B079FGPC9Y/ref=sr_1_5](url)) to the common and normally open. The switches are set to High expecting a high signal to activate. I chose high level vs. low expecting that the serial-to-parallel I2SO shift buffer would initialize in a high-z state then reset to low, which should avoid inadvertently turning on the relay. 

The dust collector is controlled via the FAN1, which is again via the I2SO circuitry. The 220V AC dust collector motor is switch through a DC to AC solid state relay such as [https://www.amazon.com/SSR-25DA-Solid-State-Output-24-380V/dp/B08FR13GYR/ref=sr_1_4](url).

Do the I2SO lines go high as part of the FluidNC startup routine?

### GCode File

GCode not loaded or running during issue.

### Other Information

```yaml
board: MKS TinyBee V1.0_001
name: CNC Router XYYZA 0-10V spindle and laser
meta: (01.02.2022) by Skorpi
stepping:
  engine: I2S_static
  idle_ms: 255
  pulse_us: 4
  dir_delay_us: 1
  disable_delay_us: 0
  segments: 12

i2so:
  bck_pin: gpio.25
  data_pin: gpio.27
  ws_pin: gpio.26

spi:
  miso_pin: gpio.19
  mosi_pin: gpio.23
  sck_pin: gpio.18

sdcard:
  cs_pin: gpio.5
  card_detect_pin: gpio.34
  frequency_hz: 8000000

kinematics:
  Cartesian:

axes:
  shared_stepper_disable_pin: NO_PIN
  shared_stepper_reset_pin: NO_PIN
  x:
    steps_per_mm: 250.000000
    max_rate_mm_per_min: 7500.000000
    acceleration_mm_per_sec2: 200.000000
    max_travel_mm: 1300.000000
    soft_limits: true
    homing:
      cycle: 3
      allow_single_axis: true
      positive_direction: true
      mpos_mm: 1265.000000
      feed_mm_per_min: 300.000000
      seek_mm_per_min: 2000.000000
      settle_ms: 250
      seek_scaler: 1.100000
      feed_scaler: 1.100000

    motor0:
      limit_neg_pin: gpio.33:low
      limit_pos_pin: NO_PIN
      limit_all_pin: NO_PIN
      hard_limits: true
      pulloff_mm: 2.500000
      standard_stepper:
        step_pin: I2SO.1
        direction_pin: I2SO.2:low
        disable_pin: I2SO.0

  y:
    steps_per_mm: 250.000000
    max_rate_mm_per_min: 7500.000000
    acceleration_mm_per_sec2: 200.000000
    max_travel_mm: 1300.000000
    soft_limits: true
    homing:
      cycle: 2
      allow_single_axis: true
      positive_direction: false
      mpos_mm: 0.000000
      feed_mm_per_min: 300.000000
      seek_mm_per_min: 2000.000000
      settle_ms: 250
      seek_scaler: 1.100000
      feed_scaler: 1.100000

    motor0:
      limit_neg_pin: gpio.32:low
      limit_pos_pin: NO_PIN
      limit_all_pin: NO_PIN
      hard_limits: true
      pulloff_mm: 2.000000
      standard_stepper:
        step_pin: I2SO.4
        direction_pin: I2SO.5:low
        disable_pin: I2SO.3

    motor1:
      limit_neg_pin: gpio.22:low
      limit_pos_pin: NO_PIN
      limit_all_pin: NO_PIN
      hard_limits: true
      pulloff_mm: 1.900000
      standard_stepper:
        step_pin: I2SO.7
        direction_pin: I2SO.8:low
        disable_pin: I2SO.6

  z:
    steps_per_mm: 250.000000
    max_rate_mm_per_min: 7500.000000
    acceleration_mm_per_sec2: 200.000000
    max_travel_mm: 300.000000
    soft_limits: false
    homing:
      cycle: 1
      allow_single_axis: true
      positive_direction: true
      mpos_mm: 0.000000
      feed_mm_per_min: 100.000000
      seek_mm_per_min: 1000.000000
      settle_ms: 500
      seek_scaler: 1.100000
      feed_scaler: 1.100000

    motor0:
      limit_neg_pin: gpio.35:low
      limit_pos_pin: NO_PIN
      limit_all_pin: NO_PIN
      hard_limits: true
      pulloff_mm: 1.500000
      standard_stepper:
        step_pin: I2SO.10
        direction_pin: I2SO.11
        disable_pin: I2SO.9

  a:
    steps_per_mm: 33.333332
    max_rate_mm_per_min: 18000.000000
    acceleration_mm_per_sec2: 1500.000000
    max_travel_mm: 325.000000
    soft_limits: false
    homing:
      cycle: 0
      allow_single_axis: true
      positive_direction: true
      mpos_mm: 0.000000
      feed_mm_per_min: 50.000000
      seek_mm_per_min: 200.000000
      settle_ms: 250
      seek_scaler: 1.100000
      feed_scaler: 1.100000

    motor0:
      limit_neg_pin: NO_PIN
      limit_pos_pin: NO_PIN
      limit_all_pin: NO_PIN
      hard_limits: false
      pulloff_mm: 1.000000
      standard_stepper:
        step_pin: I2SO.13
        direction_pin: I2SO.14
        disable_pin: I2SO.12

control:
  safety_door_pin: NO_PIN
  reset_pin: NO_PIN
  feed_hold_pin: NO_PIN
  cycle_start_pin: NO_PIN
  macro0_pin: NO_PIN
  macro1_pin: NO_PIN
  macro2_pin: NO_PIN
  macro3_pin: NO_PIN
  fault_pin: NO_PIN
  estop_pin: NO_PIN

coolant:
  flood_pin: NO_PIN
  mist_pin: I2SO.19
  delay_ms: 1000

probe:
  pin: gpio.13:low:pu
  toolsetter_pin: NO_PIN
  check_mode_start: false

macros:
  startup_line0:
  startup_line1:
  macro0:
  macro1:
  macro2:
  macro3:
  after_homing:
  after_reset:
  after_unlock:

start:
  must_home: false
  deactivate_parking: false
  check_limits: false

parking:
  enable: false
  axis: Z
  target_mpos_mm: -5.000000
  rate_mm_per_min: 800.000000
  pullout_distance_mm: 5.000000
  pullout_rate_mm_per_min: 250.000000

user_outputs:
  analog0_pin: NO_PIN
  analog1_pin: NO_PIN
  analog2_pin: NO_PIN
  analog3_pin: NO_PIN
  analog0_hz: 5000
  analog1_hz: 5000
  analog2_hz: 5000
  analog3_hz: 5000
  digital0_pin: NO_PIN
  digital1_pin: NO_PIN
  digital2_pin: NO_PIN
  digital3_pin: NO_PIN
  digital4_pin: NO_PIN
  digital5_pin: NO_PIN
  digital6_pin: NO_PIN
  digital7_pin: NO_PIN

10V:
  forward_pin: I2SO.21
  reverse_pin: NO_PIN
  pwm_hz: 5000
  direction_pin: NO_PIN
  output_pin: gpio.21
  enable_pin: NO_PIN
  disable_with_s0: true
  s0_with_disable: true
  spinup_ms: 6000
  spindown_ms: 6000
  tool_num: 0
  speed_map: 0=0.00% 4000=0.00% 24000=50.00%
  off_on_alarm: true

Laser:
  pwm_hz: 5000
  output_pin: gpio.4
  enable_pin: NO_PIN
  disable_with_s0: false
  s0_with_disable: true
  tool_num: 100
  speed_map: 0=0.00% 1000=100.00%
  off_on_alarm: true

arc_tolerance_mm: 0.002000
junction_deviation_mm: 0.010000
verbose_errors: true
report_inches: false
enable_parking_override_control: false
use_line_numbers: true
planner_blocks: 16
ok
```

$Start/Message=Grbl \V [FluidNC \B (\R) \H]
$Firmware/Build=
$SD/FallbackCS=5
$Report/Status=1
$Config/Filename=XYZAB_laser_27Nov23.yaml
$Message/Level=Debug
$WiFi/Mode=STA>AP
$Sta/SSID=RounsavallWPA2
$Sta/Password=******
$Sta/MinSecurity=WPA2-PSK
$WiFi/FastScan=OFF
$Sta/IPMode=Static
$Sta/IP=192.168.0.235
$Sta/Gateway=192.168.0.1
$Sta/Netmask=255.255.255.0
$AP/Country=01
$AP/SSID=FluidNC
$AP/Password=******
$AP/IP=192.168.0.1
$AP/Channel=1
$Hostname=fluidnc
$HTTP/BlockDuringMotion=ON
$HTTP/Enable=ON
$HTTP/Port=80
$Telnet/Enable=ON
$Telnet/Port=23
$Notification/Type=NONE
$Notification/T1=
$Notification/T2=
$Notification/TS=
ok

**Comment by bdring** [comment]
I2S chips have an indeterminate start state.

http://wiki.fluiydnc.com/en/support/controller_design_guidelines#i2so-chips

**Comment by MitchBradley** [comment]
This is the latest Tinybee schematic from the MKS Github page at https://github.com/makerbase-mks/MKS-TinyBee/blob/main/hardware/MKS%20TinyBee%20V1.0_003/MKS%20TinyBee%20V1.0_003%20SCH.pdf
![image](https://github.com/bdring/FluidNC/assets/4861133/a1cd536d-10d8-403d-a413-0ac24974bede)
As you can see, the OE_ pin is hardwired to GND, so there is no high-Z behavior at startup.  Furthermore, the chip reset SRCLR_ is pulled high so there is no hardware reset, unlike on Bart's boards where we added a capacitor to hold the reset low (active) briefly when the power first comes on, thus forcing the I2SO outputs to be initially low, at least on a power on reset.  Here is the 6 pack circuit

![image](https://github.com/bdring/FluidNC/assets/4861133/ee94dcbf-3cca-4276-af49-d18203865dc1)

FluidNC tries to set all the I2SO outputs to 0 as soon as it has discovered, via the config file, that I2SO is present on the board.  There is not a lot that the software can do when the Espressif bootloader is running, as FluidNC software does not have control of the machine then.

Perhaps you might be able to find some Tinybee pin that is at a predictable state during the problem period, and add external logic circuitry to use it disable the problem outputs then.  This is a situation over which the FluidNC firmware has very little control.  Any tricks that we might try to do to mitigate your specific situation would risk having the opposite effect on someone else.

**Comment by ColPaulR** [comment]
Thank you so much for the rapid response and the detailed explanation that my choice of control is the issue, based on their poor design choices. I'm considering using the 2nd channel of my external dual relay module as an enable signal for the dust collection solid state relay and the spindle FWD relay. I'll drive this "enable relay" in the opposite polarity as the FWD signal. If the I2SO starts up all high as it appears to do until the ESP32 can assert control then "enable relay" will prevent the signals from activating the hardware. 

Feel free to close this issue. However, I'll try to post if my new configuration mitigates the reset issue.

**Comment by ColPaulR** [comment]
I overcame this (or at least greatly reduced the frequency of occurrence) by using both channels of my dual relay module. I wired both channels of the dual relay module in series with the FWD and DCM signals (e.g. VFD FWD to COM1, NO1 to COM2 and NO2 to VFD DCM) and activate the two channels with opposite polarity outputs from the two I2SO pins (I used forward_pin: I2SO.21 and enable_pin: I2SO.22). As long as all the output pins initial to either high or low, the spindle will not spin up during ESP32 startup.
